citesphere/src/main/java/edu/asu/diging/citesphere/config/CustomAuthenticationFailureHandler.java

```diff
@@ -0,0 +1,65 @@
+package edu.asu.diging.citesphere.config;
+
+import java.io.IOException;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.MessageSource;
+import org.springframework.security.authentication.LockedException;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.web.DefaultRedirectStrategy;
+import org.springframework.security.web.RedirectStrategy;
+import org.springframework.security.web.authentication.AuthenticationFailureHandler;
+import org.springframework.security.web.util.UrlUtils;
+import org.springframework.util.Assert;
+
+public class CustomAuthenticationFailureHandler implements AuthenticationFailureHandler {
+		
+	@Autowired
+	private MessageSource messageSource;
+
+	private final Logger logger = LoggerFactory.getLogger(getClass());
+	private String defaultFailureUrl;
+	private RedirectStrategy redirectStrategy = new DefaultRedirectStrategy();
+
+	public CustomAuthenticationFailureHandler() {
+	}
+
+	public CustomAuthenticationFailureHandler(String defaultFailureUrl) {
+		setDefaultFailureUrl(defaultFailureUrl);
+	}
+
+	@Override
+	public void onAuthenticationFailure(HttpServletRequest request, HttpServletResponse response,
+			AuthenticationException exception) throws IOException, ServletException {
+
+		String errorMessage;
+
+		if (exception.getClass().equals(LockedException.class)) {
+			errorMessage = messageSource.getMessage("alert.login.user_not_approved", null, null);
+		} else {
+			errorMessage = messageSource.getMessage("alert.login.bad_credentials", null, null);
```

Comment:
move this to the template, and only use the error key here. This can get you into trouble if the error message is very long or has something in it that doesn't play nice with urls.

Reply:
and error messages are a view concern anyways.